### ISSUE TITLE
Have NodePackageInstallTask use project directory if no directory specified

### ISSUE THREAD

**Issue by mcalmus** [angular/angular-cli#21680]
# 🚀 Feature request

### Command (mark with an `x`)

- [x] new
- [ ] build
- [ ] serve
- [ ] test
- [ ] e2e
- [ ] generate
- [x] add
- [ ] update
- [ ] lint
- [ ] extract-i18n
- [ ] run
- [ ] config
- [ ] help
- [ ] version
- [ ] doc

### Description

 A clear and concise description of the problem or missing capability...

`NodePackageInstallTask` is frequently used by add tasks such as `ng-add` in `@angular/material`. If a schematic like this is called from a custom `ng-new` schematic using `externalSchematic` the install fails because it is run from some non-project directory rather than from inside the newly created project directory.

### Describe the solution you'd like

 If you have a solution in mind, please describe it.

Since schematics knows the location of `package.json` it should by default run `NodePackageInstallTask` from that same directory rather than using the working directory.

This would unfortunately qualify as a "breaking" change so a configuration option might be necessary to allow for the "old" (current) behavior once this change is made.

### Describe alternatives you've considered

 Have you considered any alternative solutions or workarounds?
The only work-around I've been able to do is essentially replicating the behaviors of third-party schematic "add" tasks within my custom schematic and then running install after the fact which is less than ideal.

**Comment by mcalmus** [comment]
Related to https://github.com/angular/components/issues/23436

**Comment by alan-agius4** [comment]
Can you please share a snipper of the custom `ng-new` schematic, is a workspace being created?

**Comment by mcalmus** [comment]
Yes. A workspace is being created. Here is a snipped of the chain of calls.
```
  return chain([
    mergeWith(
      apply(empty(), [
        externalSchematic('@schematics/angular', 'workspace', workspaceOptions),
        externalSchematic('@schematics/angular', 'application', applicationOptions),
        ....
        options.skipMaterial ? noop() : externalSchematic('@angular/material', 'ng-add', materialInstallOptions),
        ...
```

**Comment by mcalmus** [comment]
Following up. Is the provided snippet sufficient?

**Comment by alan-agius4** [comment]
Yes it is.

**Comment by mcalmus** [comment]
Could this fix similarly be used here? https://github.com/angular/angular-cli/commit/db0cb5593dcde395f1877f7c9b6bdf911cd26654
